src/python_testing/TestConformanceSupport.py

```diff
@@ -0,0 +1,564 @@
+
+        # none, AB, CD, AB&CD
+        feature_maps = [0x00, 0x01, 0x02, 0x03]
+        has_ab = [False, True, False, True]
+        has_cd = [False, False, True, True]
+
+        attribute_names_to_values = {'attr1': 0x00, 'attr2': 0x01}
+        attribute_lists = [[], [0x00], [0x01], [0x00, 0x01]]
+        has_attr1 = [False, True, False, True]
+        has_attr2 = [False, False, True, True]
+
+        command_names_to_values = {'cmd1': 0x00, 'cmd2': 0x01}
+        cmd_lists = [[], [0x00], [0x01], [0x00, 0x01]]
+        has_cmd1 = [False, True, False, True]
+        has_cmd2 = [False, False, True, True]
+
+        params = ConformanceParseParameters(
+            feature_map=feature_names_to_bits, attribute_map=attribute_names_to_values, command_map=command_names_to_values)
+
+        # Start simple - mandatory, optional, disallowed
+        xml = '<mandatoryConform />'
+        et = ElementTree.fromstring(xml)
+        callable = parse_callable_from_xml(et, params)
+        for f in feature_maps:
+            asserts.assert_equal(callable(f, [], []), ConformanceDecision.MANDATORY)
+
+        xml = '<optionalConform />'
+        et = ElementTree.fromstring(xml)
+        callable = parse_callable_from_xml(et, params)
+        for f in feature_maps:
+            asserts.assert_equal(callable(f, [], []), ConformanceDecision.OPTIONAL)
+
+        xml = '<disallowConform />'
+        et = ElementTree.fromstring(xml)
+        callable = parse_callable_from_xml(et, params)
+        for f in feature_maps:
+            asserts.assert_equal(callable(f, [], []), ConformanceDecision.DISALLOWED)
+
+        xml = '<deprecateConform />'
+        et = ElementTree.fromstring(xml)
+        callable = parse_callable_from_xml(et, params)
+        for f in feature_maps:
+            asserts.assert_equal(callable(f, [], []), ConformanceDecision.DISALLOWED)
+
+        xml = '<provisionalConform />'
+        et = ElementTree.fromstring(xml)
+        callable = parse_callable_from_xml(et, params)
+        for f in feature_maps:
+            asserts.assert_equal(callable(f, [], []), ConformanceDecision.PROVISIONAL)
+
+        # single feature mandatory
+        xml = '<mandatoryConform>\n'
+        xml = xml + '<feature name="AB" />\n'
+        xml = xml + '</mandatoryConform>'
+        et = ElementTree.fromstring(xml)
+        callable = parse_callable_from_xml(et, params)
+        for i, f in enumerate(feature_maps):
+            if has_ab[i]:
+                asserts.assert_equal(callable(f, [], []), ConformanceDecision.MANDATORY)
+            else:
+                asserts.assert_equal(callable(f, [], []), ConformanceDecision.NOT_APPLICABLE)
+
+        xml = '<mandatoryConform>\n'
+        xml = xml + '<feature name="CD" />\n'
+        xml = xml + '</mandatoryConform>'
+        et = ElementTree.fromstring(xml)
+        callable = parse_callable_from_xml(et, params)
+        for i, f in enumerate(feature_maps):
```

Comment:
nit: these are a lot of seemingly separate tests. Instead of having one large test, could we split them?